### ISSUE TITLE
Removing a column with ComponentRenderer caused an exception

### ISSUE THREAD

**Issue by lorenzobenvenuti** [datenhahn/componentrenderer#18]
Hi,

trying to remove a column that is using a ComponentRenderer produces an exception:

```
java.lang.IllegalStateException: Renderer is not attached to any parent
    at com.vaadin.ui.Grid$AbstractGridExtension.getParentGrid(Grid.java:4366)
    at de.datenhahn.vaadin.componentrenderer.ComponentRenderer.setParent(ComponentRenderer.java:76)
    at com.vaadin.server.AbstractClientConnector.removeExtension(AbstractClientConnector.java:597)
    at com.vaadin.ui.Grid.internalRemoveColumn(Grid.java:5271)
    at com.vaadin.ui.Grid.removeColumn(Grid.java:5260)
```

_AbstractClientConnector.removeExtension_ invokes _extension.setParent(null)_; _ComponentRenderer.setParent_ does:

```
super.setParent(parent);
// VERY IMPORTANT: registers the DataGenerator extension
extend(getParentGrid());
```

But _getParentGrid_ throws an exception if parent is null! I've seen a previous version of the class was checking if parent was set to null before invoking _extend(getParentGrid())_...

Thank you,

lorenzo

**Comment by datenhahn** [comment]
Thanks for reporting that issue, I must have accidentally removed the null check. It is fixed and deployed now (1.0.0beta4). I also wrote a test so, it doesn't creep in again :)

**Comment by lorenzobenvenuti** [comment]
Thank you!
